Include/pythonrun.h

```diff
@@ -25,8 +25,13 @@ PyAPI_DATA(int) (*PyOS_InputHook)(void);
  * no two calls to check recursion depth are more than this far
  * apart. In practice, that means it must be larger than the C
  * stack consumption of PyEval_EvalDefault */
```

Comment:
This seems sort of fragile? What if there's a ton of unrelated C calls (maybe from a C extension) between two interpreter entries? Does that cause this to be incorrect?

Reply:
Any C code that can recurse needs to do checks.
We already do that in plenty of places, like the parser, compiler, json encoding, etc.
C extensions should do it too.